AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -360,17 +360,23 @@ - (BOOL)removeAccount {
   return YES;
 }
 
-- (void)acquireTokenSilentlyWithMSALAccount:(MSALAccount *)account {
+- (void)acquireTokenSilentlyWithMSALAccount:(MSALAccount *)account withUIFallback:(BOOL)withUIFallback {
   __weak typeof(self) weakSelf = self;
   [self.clientApplication
       acquireTokenSilentForScopes:@[ (NSString * __nonnull) self.authConfig.authScope ]
                           account:account
-                  completionBlock:^(MSALResult *result, NSError *e) {
+                  completionBlock:^(MSALResult *result, NSError *error) {
                     typeof(self) strongSelf = weakSelf;
-                    if (e) {
-                      MSLogWarning([MSAuth logTag],
-                                   @"Silent acquisition of token failed with error: %@. Triggering interactive acquisition", e);
-                      [strongSelf acquireTokenInteractively];
+                    if (error) {
+                      if (withUIFallback && [error.domain isEqual:MSALErrorDomain] && error.code == MSALErrorInteractionRequired) {
+                        MSLogWarning([MSAuth logTag],
```

Comment:
`Info` level?

Reply:
Since `MSALErrorInteractionRequired` failure of silent sign-in is kind of expected when user leaves the app for too long, so warning is probably too much and `Info` level is enough.